Controller/Admin/ShopMain.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * This Software is the property of OXID eSales and is protected
+ * by copyright law - it is NOT Freeware.
+ *
+ * Any unauthorized use of this software without a valid license key
+ * is a violation of the license agreement and will be prosecuted by
+ * civil and criminal law.
+ *
+ * @category      module
+ * @author        OXID Professional services
+ * @link          http://www.oxid-esales.com
+ * @copyright (C) OXID eSales AG 2003-2020
+ */
+
+namespace OxidProfessionalServices\ModulesConfig\Controller\Admin;
+
+use OxidEsales\Eshop\Application\Controller\Admin\ShopMain as ShopMainCore;
+
+/**
+ * ShopMain.
+ */
+class ShopMain extends ShopMainCore
+{
+    /**
+     * Check if Shop can be created.
+     *
+     * @param string                                   $shopId
+     * @param \OxidEsales\Eshop\Application\Model\Shop $shop
+     *
+     * @return bool
+     */
+    public function canCreateShop($shopId, $shop)
+    {
+        return parent::canCreateShop($shopId, $shop);
```

Comment:
I don't get the point of this class

Reply:
Ahh, Sorry i missed your comment. 

Actually i want to call **canCreateShop** method in our command. but this method is protected. Hence, i am just overriding this method and making it public. If you have any better suggestion then please let me know. 
phttps://github.com/OXID-eSales/oxideshop_ce/blob/master/source/Application/Controller/Admin/ShopMain.php#L290
https://github.com/OXID-eSales/oxideshop_ee/blob/master/Application/Controller/Admin/ShopMain.php#L121